packages/airnode-node/src/types.ts

```diff
@@ -9,6 +7,43 @@ import {
   RequestedWithdrawalEvent,
   FulfilledWithdrawalEvent,
 } from '@api3/airnode-protocol';
+// TODO: Make this nicer
```

Comment:
@amarthadan Suggested importing the types directly from validator, which is a better approach to re-exporting.

However, the inferred schemas from zod have a disadvantage that you don't really see the type. For example, when looking at the `Config` type here, you will see the following:
```
export declare type Config = SchemaType<typeof configSchema>;
```

ofc, the `configSchema` is in the same file, but it looks like this:
```
export declare const configSchema: z.ZodObject<{
    chains: z.ZodArray<z.ZodObject<{
        authorizers: z.ZodArray<z.ZodString, "many">;
        blockHistoryLimit: z.ZodOptional<z.ZodNumber>;
        contracts: z.ZodObject<{
            AirnodeRrp: z.ZodString;
        }, "strip", z.ZodTypeAny, {
            AirnodeRrp: string;
        }, {
            AirnodeRrp: string;
        }>;
        id: z.ZodString;
        ignoreBlockedRequestsAfterBlocks: z.ZodOptional<z.ZodNumber>;
        minConfirmations: z.ZodOptional<z.ZodNumber>;
        type: z.ZodLiteral<"evm">;
        options: z.ZodObject<{
            txType: z.ZodUnion<[z.ZodLiteral<"legacy">, z.ZodLiteral<"eip1559">]>;
            baseFeeMultiplier: z.ZodOptional<z.ZodString>;
            priorityFee: z.ZodOptional<z.ZodObject<{
                value: z.ZodString;
                unit: z.ZodOptional<z.ZodUnion<[z.ZodLiteral<"wei">, z.ZodLiteral<"kwei">, z.ZodLiteral<"mwei">, z.ZodLiteral<"gwei">, z.ZodLiteral<"szabo">, z.ZodLiteral<"finney">, z.ZodLiteral<"ether">]>>;
            }, "strip", z.ZodTypeAny, {
                unit?: "wei" | "kwei" | "mwei" | "gwei" | "szabo" | "finney" | "ether" | undefined;
                value: string;
            }, {
                unit?: "wei" | "kwei" | "mwei" | "gwei" | "szabo" | "finney" | "ether" | undefined;
                value: string;
            }>>;
        }, "strip", z.ZodTypeAny, {
            baseFeeMultiplier?: string | undefined;
            priorityFee?: {
                unit?: "wei" | "kwei" | "mwei" | "gwei" | "szabo" | "finney" | "ether" | undefined;
                value: string;
            } | undefined;
            txType: "legacy" | "eip1559";
        }, {
...
```

I think the better solution is to write those manually here - the zod schema will be compatible (but we will have to update two locations when making changes to config.json). I also thought about generating these schemas, but that has the same problem as with inferring (TS is not able to extract subtypes). 

Any better ideas?

Reply:
I don't agree. I would still prefer keeping the types in one place, updated automatically when the schema is updated. I agree that the generated types are a bit harder to read but it's still quite straightforward. In my opinion, the advantage of having the types in one place and automatically updated with the schema outweighs the readability issue.

Reply:
> but it's still quite straightforward

I disagree. Take a look at `chains.options` and try to understand what is its type.

> the advantage of having the types in one place

This is a good point, and we check only that zod schema is compatible with the manually defined one, but not the check other way around (because it doesn't hold, since zod types are more precise). This means that inconsistencies can occur. However, we will likely notice, because we will be missing the functionality in Airnode.

> and automatically updated with the schema outweighs the readability issue.

To me not, because the schema is updated only from time to time.

Reply:
I will keep this comment unresolved and we can discuss that in the google document.